lib/pigeon/registry.ex

```diff
@@ -20,9 +20,7 @@ defmodule Pigeon.Registry do
   def next(pid) do
     __MODULE__
     |> Registry.lookup(pid)
-    |> case do
-      [] -> nil
-      pids -> pids |> Enum.random() |> elem(0)
-    end
+    |> Enum.min_by(fn {_, priority} -> priority end, fn -> {nil, 0} end)
```

Comment:
```suggestion
    |> Enum.shuffle()
    |> Enum.min_by(fn {_, priority} -> priority end, fn -> {nil, 0} end)
```
Probably not a big issue if this value is being constantly updated, but should make it so that even if all hosts have around the same priority, we won't get the same pid back each time.